README.md

```diff
@@ -58,50 +67,137 @@ const colors = require(`${themeDirRequire}/tailwindcss/colors`);
+instead of adding all of the logic needed your self, these functions will handle this for you.
+
+with `twVar` you can add a CSS variable to one or more TailwindCSS tokens, for example like this:
+
+```js
+const { twVar, mergeTailwindConfig } = require('@hyva-themes/hyva-modules');
+
+module.exports = mergeTailwindConfig({
+    theme: {
+        extend: {
+            colors: {
+                primary: {
+                    lighter: twVar('primary-lighter', colors.blue["600"]),
+                    DEFAULT: twVar('primary', colors.blue["700"]),
+                    darker: twVar('primary-darker', colors.blue["800"]),
+                },
+            }
+        }
+    }
+    // The rest of your tailwind config...
+})
+```
+
+this will render any Tailwind color utility class with the following CSS value:
+
+```css
+.bg-primary {
+    --tw-bg-opacity: 1;
+    background-color: color-mix(in srgb, var(--color-primary, #1d4ed8) calc(var(--tw-bg-opacity) * 100%), transparent);
+}
+```
+
+> The method for handling opacity with all color syntaxes is based on the upcoming TailwindCSS v4,
+> here they use the CSS function `color-mix()` to make this possible,
+> we have reused this to make that transition easier in the future.
+
+and with this you can change the value in your CSS or inline in the page with:
+
+```css
+:root {
+    --color-primary: hsl(20 80% 50%);
+}
+```
+
+Now if you don't want to set this for each TailwindCSS token we also offer the functions `twProps`.
+This acts as a wrapper and will use the keys as the name for the CSS variable,
+if we use the same example we can create the same effect with less effort:
+
+```js
+const { twProps, mergeTailwindConfig } = require('@hyva-themes/hyva-modules');
+
+module.exports = mergeTailwindConfig({
+    theme: {
+        extend: {
+            colors: twProps({
+                primary: {
+                    lighter: colors.blue["600"],
+                    DEFAULT: colors.blue["700"],
+                    darker: colors.blue["800"],
+                },
+            }),
+            textColors: {
+                primary: twProps({
+                    lighter: colors.blue["600"],
+                    DEFAULT: colors.blue["700"],
+                    darker: colors.blue["800"],
+                }, 'text-primary'),
+            }
+        }
+    }
+    // The rest of your tailwind config...
+})
+```
+
+**Important to note:**
+
+- `twVar` only works with string values.
+- `twProps` can be used on any level of your Tailwind config, but it only works with string values.
+- `twProps` is best used inside the [reference function method](https://tailwindcss.com/docs/theme#referencing-other-values).
+- You can use `Object.assign()` to split 2 groups inside any Tailwind config group, e.g. `colors`, so only one part gets the variables and the other part is left as is.
+- Both `twVar` and `twProps` have a prefix argument, this also highlighted when using Intellisense in your editor, powered trough JSDoc.
```

Comment:
This also needs an example. The " powered trough JSDoc" can be removed as it is an implementation detail.